### PR TITLE
feature/AT-RSC-UI-ListStory

### PR DESCRIPTION
**Pull request by Yairmar**

### CODE REVIEW

File: src/services/at-resource-api.service.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+const AT_RESOURCES_SERVICE_URI = process.env.AT_RESOURCES_SERVICE_URI;
+console.log("AT_UNIVERSITY_SERVICE_URI: " + AT_RESOURCES_SERVICE_URI);
+
+const resourceServiceApi = {};
+
+resourceServiceApi.getStoryList = () => {
+    return axios({
+      method: "GET",
+      url: AT_RESOURCES_SERVICE_URI + `/v1/course`,
```
Comment (fasaga) [inline src/services/at-resource-api.service.js]: change course to story

File: src/controllers/admin.controller.js [hunk 2]
```diff
@@ -2,11 +2,30 @@ const adminCtrl = {};
...
 
 // AT-RESOURCES - Admin - Index
 adminCtrl.renderIndex = async(req, res) => {
     console.log("--> adminCtrl.renderIndex");
     res.render("admin/index");
 };
 
+// AT-RESOURCES - Admin - Story list
+adminCtrl.renderStoryList = async (req, res) => {
+    let story = [];
```
Comment (fasaga) [inline src/controllers/admin.controller.js]: name this variable in plural, story -> stories

File: src/views/admin/story/list-story.hbs [hunk 3]
```diff
@@ -0,0 +1,60 @@
+
+<div class="tab-content card" id="nav-tabContent">
+    <div class="tab-pane fade show active" id="search-md" role="tabpanel" aria-labelledby="search-tab-md">
+        <!-- Course Search -->
+        <section class="card">
+            <div class="card-body">
+                <table style="width: 100%">
+                    <tr>
+                        <td style="text-align: left;">
+                            <h4 class="title_section">Courses Module</h4>
```
Comment (fasaga) [inline src/views/admin/story/list-story.hbs]: change this title to Stories

File: src/views/admin/story/list-story.hbs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                        </td>
+                        <td style="text-align: right;">
+                            <a class="btn btn-primary" href="/admin/story/add/">&raquo; Add New Course</a>
```
Comment (fasaga) [inline src/views/admin/story/list-story.hbs]: change the title to Add New Story

File: src/views/admin/story/list-story.hbs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                        </td>
+                    </tr>
+                </table>
+            </div>
+        </section>
+
+        <section class="card">
+            <div class="card-body">
+                <table id="coursesTable" class="display responsive nowrap" style="width:100%">
```
Comment (fasaga) [inline src/views/admin/story/list-story.hbs]: rename the id to storiesTable

File: src/views/admin/story/list-story.hbs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                    <tbody>
+                        {{#each courses}}
+                            <tr>
+                                <td>{{_id}}</th>
+                                <td>{{category}}</th>
+                                <td>{{title}}</td>
+                                <td>{{description}}</td>
+                                <td>{{status}}</td>
+                                <td>
+                                    <a href="/admin/course/edit/{{_id}}"><i class="fas fa-edit"></i></a>
```
Comment (fasaga) [inline src/views/admin/story/list-story.hbs]: update these urls

File: src/views/admin/story/list-story.hbs [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+                                    <a href="/admin/course/delete/{{_id}}"><i class="fas fa-trash" style="color:red"></i></a>
+                                </td>
+                            </tr>
+                        {{/each}}
+                    </tbody>
+                </table>
+                <script type="text/javascript">
+                    $(document).ready(function () {
+                        $('#coursesTable').DataTable();
```
Comment (fasaga) [inline src/views/admin/story/list-story.hbs]: use storiesTable

File: src/views/admin/story/list-story.hbs [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+                            <th>ID</th>
+                            <th>Category</th>
+                            <th>Title</th>
+                            <th>Description</th>
+                            <th>Status</th>
+                            <th>Actions</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {{#each courses}}
```
Comment (fasaga) [inline src/views/admin/story/list-story.hbs]: stories instead of courses

File: src/views/admin/story/list-story.hbs [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+        </section>
+
+        <section class="card">
+            <div class="card-body">
+                <table id="coursesTable" class="display responsive nowrap" style="width:100%">
+                    <thead>
+                        <tr>
+                            <th>ID</th>
+                            <th>Category</th>
+                            <th>Title</th>
```
Comment (fasaga) [inline src/views/admin/story/list-story.hbs]: change title to name

File: src/views/admin/story/list-story.hbs [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+                            <th>Status</th>
+                            <th>Actions</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {{#each courses}}
+                            <tr>
+                                <td>{{_id}}</th>
+                                <td>{{category}}</th>
+                                <td>{{title}}</td>
```
Comment (fasaga) [inline src/views/admin/story/list-story.hbs]: change title to name

File: src/views/admin/story/list-story.hbs [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+            </div>
+        </section>
+
+        <section class="card">
+            <div class="card-body">
+                <table id="coursesTable" class="display responsive nowrap" style="width:100%">
+                    <thead>
+                        <tr>
+                            <th>ID</th>
+                            <th>Category</th>
```
Comment (fasaga) [inline src/views/admin/story/list-story.hbs]: use userName instead of category

File: src/views/admin/story/list-story.hbs [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+                            <th>Description</th>
+                            <th>Status</th>
+                            <th>Actions</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {{#each courses}}
+                            <tr>
+                                <td>{{_id}}</th>
+                                <td>{{category}}</th>
```
Comment (fasaga) [inline src/views/admin/story/list-story.hbs]: use userName instead of category

File: src/controllers/admin.controller.js [hunk 13]
```diff
@@ -79,7 +67,7 @@ adminCtrl.deleteStory = (req, res) => {
 adminCtrl.renderEditStoryForm = async(req, res) => {
```
Comment (at-internship) [inline src/controllers/admin.controller.js]: add this one after renderStoryList

File: src/controllers/admin.controller.js [hunk 14]
```diff
@@ -89,4 +77,22 @@ adminCtrl.deleteStory = async(req, res) => {
...
+        if (responseStoryList === null || responseStoryList === undefined){
+            req.flash("error_msg", "Service unavailable");
+        } else {
+            console.log("--> adminCtrl.renderStoryList");
+            stories = responseStoryList.data;
+        }
+    } catch(err) {
+        console.error(err.message);
+    }  finally {
+        res.render("admin/story/list-story", { stories});
```
Comment (at-internship) [inline src/controllers/admin.controller.js]: it was res.render("admin/story/index", { stories});

**Review by at-internship — CHANGES_REQUESTED** [review]

File: src/controllers/admin.controller.js (hunk 14 above)
Comment (Efcobr-AT) [reply src/controllers/admin.controller.js]: one question, in the direcory (story) exits the file index.hbs? or is rename the file list-story.hbs for index.hbs and render to this. Or is render to file index the directory admin.

File: src/controllers/admin.controller.js (hunk 13 above)
Comment (Efcobr-AT) [reply src/controllers/admin.controller.js]: by this, it did refers the function renderEditStoryForm

File: src/controllers/admin.controller.js [hunk 15]
```diff
@@ -63,36 +81,4 @@ adminCtrl.deleteStory = (req, res) => {
     res.redirect("/admin/Story");
 };
 
-// AT-RESOURCES - Admin - Render Edit Story Form
-adminCtrl.renderEditStoryForm = async(req, res) => {
```
Comment (at-internship) [inline src/controllers/admin.controller.js]: Why did you remove this function?

File: src/controllers/admin.controller.js [hunk 16]
```diff
@@ -63,36 +81,4 @@ adminCtrl.deleteStory = (req, res) => {
...
-    console.log("--> adminCtrl.renderEditStoryForm");
-    res.render("admin/story/edit-story");
-}
-// AT-RESOURCES - Admin - Delete Story
-adminCtrl.deleteStory = async(req, res) => {
```
Comment (at-internship) [inline src/controllers/admin.controller.js:92]: Why did you remove this function?

File: src/routes/admin.routes.js [hunk 17]
```diff
@@ -3,7 +3,7 @@ const router = express.Router();
 const path = require('path');
 
 // Admin Controller
-const { renderIndex,
+const { renderIndexAdmin,
         renderStoryList,
         renderAddStoryForm,
         addStory,
```
Comment (at-internship) [inline src/routes/admin.routes.js:9]: Add renderEditStoryForm, deleteStory

**Review by at-internship — CHANGES_REQUESTED** [review]

File: src/controllers/admin.controller.js (hunk 16 above)
Comment (Efcobr-AT) [reply src/controllers/admin.controller.js:92]: it is duplicated:
https://github.com/at-internship/at-resources-ui/commit/2f9c8c7535e913c24ab1f456754f17dd96e67546
But if it exists

File: src/controllers/admin.controller.js (hunk 15 above)
Comment (Efcobr-AT) [reply src/controllers/admin.controller.js]: it is duplicated:
https://github.com/at-internship/at-resources-ui/commit/2f9c8c7535e913c24ab1f456754f17dd96e67546
But if it exists

File: src/routes/admin.routes.js (hunk 17 above)
Comment (Efcobr-AT) [reply src/routes/admin.routes.js:9]: are added

File: src/controllers/admin.controller.js (hunk 15 above)
Comment (fasaga) [reply src/controllers/admin.controller.js]: was duplicated in lines 56,62

File: src/controllers/admin.controller.js (hunk 16 above)
Comment (fasaga) [reply src/controllers/admin.controller.js:92]: duplicated in lines 70,76

File: src/routes/admin.routes.js (hunk 17 above)
Comment (fasaga) [reply src/routes/admin.routes.js:9]: lines 10 and 12

**Review by fasaga — APPROVED** [review]

**Review by at-internship — APPROVED** [review]
Approved

**Review by Efcobr-AT — APPROVED** [review]

**Review by MajoYanez — APPROVED** [review]

File: src/views/admin/story/index.hbs [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+            </div>
+        </section>
+
+        <section class="card">
+            <div class="card-body">
+                <table id="storiesTable" class="display responsive nowrap" style="width:100%">
+                    <thead>
+                        <tr>
+                            <th>ID</th>
+                            <th>Name</th>
```
Comment (at-internship) [inline src/views/admin/story/index.hbs:27]: These are the properties of the "Story" collection

```
id - Object ID
sprint_id - Object ID
user_id - Object ID
prority - Integer (1 - High, 2 - Medium, 3 - Low)
name - String
description - String
aceptance_criteria - String
story_points - Integer (1, 3, 5)
progress - Integer
start_date - Date
due_date - Date
create_date - Date
status - Integer (0, 1)
```

**Review by at-internship — DISMISSED** [review]

**Review by LuisAT07 — APPROVED** [review]

**Review by fasaga — APPROVED** [review]
